File: 雄安路网/xiongan_v5.1/backend/app/api/events.py
from fastapi import APIRouter, Body, Request

from app.api.common import ok

router = APIRouter(prefix="/events", tags=["events"])


@router.post("/inject")
async def inject(request: Request, body: dict = Body(...)):
    return ok(request.app.state.runtime.inject_event(
        body.get("event_type", ""), body.get("params") or {}))


@router.get("/list")
async def list_events(request: Request):
    return ok(request.app.state.runtime.list_events())

File: 雄安路网/xiongan_v5.1/frontend/src/stores/events.js
import { defineStore } from 'pinia'
import { apiGet, apiPost } from '../api/http'

/** 扰动事件 */
export const useEventStore = defineStore('events', {
  state: () => ({ list: [] }),
  actions: {
    async refresh() {
      try { this.list = await apiGet('/events/list') } catch { /* 静默 */ }
    },
    async inject(eventType, params = {}) {
      await apiPost('/events/inject', { event_type: eventType, params })
      await this.refresh()
    },
  },
})
